scripts/utilities/retry.js

```diff
@@ -1,20 +1,22 @@
-const retry = (functionToTry, maxAttempts = 3, delay = 1000) => {
-  return new Promise((resolve, reject) => {
-    let remainingAttempts = maxAttempts;
-    const intervalId = setInterval(() => {
-      try {
-        functionToTry();
-        clearInterval(intervalId);
-        resolve();
-      } catch (error) {
-        remainingAttempts -= 1;
-        if (remainingAttempts <= 0) {
-          clearInterval(intervalId);
-          reject(error);
-        }
+function timeout(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+function retry(functionToTry, maxAttempts = 3, delay = 1000) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      functionToTry();
+      resolve();
+    } catch (error) {
+      const remainingAttempts = maxAttempts - 1;
+      if (remainingAttempts <= 0) {
+        reject(error);
+      } else {
+        await timeout(delay);
+        retry(functionToTry, remainingAttempts, delay);
```

Comment:
Did you make this that the retry logic works here? There's a weird interaction here and I'm not sure how things interact

If the execution ends up in this `else` block then `resolve()` never gets called. Does that mean that the promise never resolves? I'm not sure

Reply:
Good point i'll fix that up!